test/lib/testMayaUsdProxyAccessor.py

```diff
@@ -0,0 +1,1265 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import tempfile
+import ufe
+import unittest
+
+import maya.cmds as cmds
+
+from mayaUsd import lib as mayaUsdLib
+from mayaUsd.lib import GetPrim
+from mayaUsd.lib import proxyAccessor as pa
+
+from pxr import Usd, UsdGeom, Sdf, Tf, Vt
+
+from maya.debug.emModeManager import emModeManager
+from maya.plugin.evaluator.CacheEvaluatorManager import CacheEvaluatorManager, CACHE_STANDARD_MODE_EVAL
+
+class NonCachingScope(object):
```

Comment:
All these helper classes and methods will have to be refactored. It's not part of this PR and we have another PR that moves tests out of Pixar plugin into our global test folder. We will coordinate on where test utils for caching and UFE should live after this PR.